app/presenters/users/friends_presenter.rb

```diff
@@ -23,6 +27,19 @@ def show_device
       @device = @friend.devices.where(cloaked: false).find(@params[:device_id])
     end
 
+    def form_for
+      @friend
+    end
+
+    def form_path
+      show_device_user_friend_path(@user.url_id, @friend, device_id: @device.id)
```

Comment:
since there is attr reader for :device, there is no need to use instance variable